pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -856,6 +858,32 @@ func (r *ReconcileSyncSetInstance) clearUnknownObjectSyncCondition(syncSetCondit
 	)
 }
 
+// filterApplyError determines if the apply error contains the applyTempFile prefix and
+// trims the beginning of the error message removing everything up to the temporary path
+// so that we only return the useful part of the error message.
+// The message we set in the apply failed status condition needs to remain static so that
+// we don't update with a new message every reconcile as condition updates rely on
+// controllerutils.UpdateConditionIfReasonOrMessageChange.
+// Ex:
+//     applyErr="error when creating \"/tmp/apply-475927931\": namespaces \"openshift-am-config\" not found"
+// Returns:
+//     "namespaces \"openshift-am-config\" not found"
+func filterApplyError(applyErr string) string {
+	if !strings.Contains(applyErr, applyTempFile) {
+		return applyErr
+	}
+
+	splitErr := strings.Split(applyErr, " ")
```

Comment:
How would you feel about just replacing the filename with a constant string like "file"? Or do you explicitly want to remove the "error when creating" part, too?

See https://play.golang.org/p/m3UreAoTzG2

Reply:
If you did want to remove the "error when creating" part, then I suggest something like the following.

https://play.golang.org/p/zgvVEm2-K9X

My concern is that splitting and rejoining is not very efficient.

Reply:
I updated it to remove the "error when creating" part using your second example as to me it seems related to creating the contents of the tempfile which is extra information as far as applying goes.